src/clj/rems/workflow/dynamic.clj

```diff
@@ -827,6 +902,61 @@
                               [{:type ::invite-member :actor "assistant" :member {:name "Member Applicant 1" :email "member1@applicants.com"}}]
                               injections)))))))
 
+(deftest test-remove-member
+  (let [application {:state ::submitted
+                     :members [{:userid "applicant"} {:userid "somebody"}]
+                     :applicantuserid "applicant"
+                     :workflow {:type :workflow/dynamic
+                                :handlers ["assistant"]}}
+        injections {}]
+    (testing "remove member by applicant"
+      (is (= [{:userid "applicant"}]
+             (:members
+              (apply-commands application
+                              [{:type ::remove-member :actor "applicant" :member {:userid "somebody"}}]
+                              injections)))))
+    (testing "remove applicant by applicant"
+      (is (= {:errors [{:type :cannot-remove-applicant}]}
+             (handle-command {:type ::remove-member :actor "applicant" :member {:userid "applicant"}}
+                             application
+                             injections))))
+    (testing "remove member by handler"
+      (is (= [{:userid "applicant"}]
+             (:members
+              (apply-commands application
+                              [{:type ::remove-member :actor "assistant" :member {:userid "somebody"}}]
+                              injections)))))
+    (testing "only members can be removed"
+      (is (= {:errors [{:type :user-not-member :user {:userid "notamember"}}]}
+             (handle-command {:type ::remove-member :actor "assistant" :member {:userid "notamember"}}
+                             application
+                             injections))))))
+
+(deftest test-uninvite-member
+  (let [application {:state ::submitted
+                     :invited-members [{:name "Some Body" :email "some@body.com"}]
+                     :applicantuserid "applicant"
+                     :workflow {:type :workflow/dynamic
+                                :handlers ["assistant"]}}
```

Comment:
Please refactor this so that the application is produced from events with `apply-events`, similar to all the other tests.

Reply:
Ok I based my tests on earlier code that didn't use the events.